src/Simulation/NativeSparseSimulator/capi.cpp

```diff
@@ -52,45 +52,56 @@ extern "C"
         return getSimulator(sim_id)->get_num_qubits();
     }
 
-// Generic single-qubit gate
-#define FWDGATE1(G)                                                                                                    \
-    MICROSOFT_QUANTUM_DECL void G##_cpp(simulator_id_type sim_id, logical_qubit_id q)                                                   \
-    {                                                                                                                  \
-        getSimulator(sim_id)->G(q);                                                                  \
+    // single-qubit gates
+    MICROSOFT_QUANTUM_DECL void X_cpp(simulator_id_type sim_id, logical_qubit_id q)
+    {
+        getSimulator(sim_id)->X(q);
```

Comment:
I don't insist, but these changes are hardly related to the problem being fixed.

Reply:
Agreed. I just fixed stuff that prevented me from debugging code efficiently. I can move them to a separate PR.